osquery/core/init.cpp

```diff
@@ -97,13 +101,20 @@ enum {
 namespace {
 extern "C" {
 static inline bool hasWorkerVariable() {
-  return (getenv("OSQUERY_WORKER") != nullptr);
+  auto value = ::osquery::getEnvVar("OSQUERY_WORKER");
+  if (value) {
```

Comment:
Will that work for `boost::optional`? Would explicitly casting it to a `bool` make it more clear as to the intention of the code?